### DISCUSSION TITLE
Not 100% a bug but interaction with sweepers and queue going wrong

### DISCUSSION THREAD

**Discussion by ReinSonozaki** [c0nD/FluffyBOT#49] [Bug Report]
What happened was there are 3 ppl 
A hitting 
B & C in queue 

A is done and sweeper is pinged, so what happens is B is pulled out of the queue and is said to go next 
but when the sweeper is done with the sweep, it then says its C turn, as it doesnt check if sweeper and B are the same person

**Comment by c0nD** [comment]
An idea that I've had to resolve that would possibly to modify the done message for possible sweeper commands (`/killed` and `resume_hit`).

Another idea would be to do some sort of sweep commands or a "command menu" for sweepers.

Either way I think there needs to be some sort of distinction to where it will say that they're a sweeper and it won't set the `done` flag to true, and therefor won't "push" people along in the queue.

@OneStone2 -- I'm curious as to if you have any thoughts about how to resolve this? If anyone else has ideas of what might be easiest for them, please add to the discussion.

**Reply by OneStone2** [reply]
simplest way to fix it imo is to only advance the queue to C whenever B hits, bc this might not only apply to sweeps

**Comment by c0nD** [comment]
Old issue -- causation from another issue.

Closed.
